sdk/messaging/azservicebus/internal/namespace.go

```diff
@@ -249,47 +240,46 @@ func (ns *Namespace) Close(ctx context.Context) error {
 	return nil
 }
 
-// Recover destroys the currently held client and recreates it.
-// clientRevision being nil will recover without a revision check.
-func (ns *Namespace) Recover(ctx context.Context, clientRevision uint64) error {
+// Recover destroys the currently held AMQP connection and recreates it, if needed.
+// If a new is actually created (rather than just cached) then the returned bool
+// will be true. Any links that were created from the original connection will need to
+// be recreated.
+func (ns *Namespace) Recover(ctx context.Context, theirConnID uint64) (bool, error) {
 	ns.clientMu.Lock()
 	defer ns.clientMu.Unlock()
 
 	_, span := tab.StartSpan(ctx, tracing.SpanRecoverClient)
 	defer span.End()
 
 	span.AddAttributes(
-		tab.Int64Attribute("revision", int64(ns.clientRevision)),
-		tab.Int64Attribute("requested", int64(clientRevision)))
+		tab.Int64Attribute("connID", int64(ns.connID)),
+		tab.Int64Attribute("theirConnID", int64(theirConnID)))
 
-	if ns.clientRevision > clientRevision {
-		span.Logger().Info(fmt.Sprintf("Skipping recovery, already recovered: %d vs %d", ns.clientRevision, clientRevision))
+	if ns.connID != theirConnID {
+		log.Writef(EventConn, "Skipping connection recovery, already recovered: %d vs %d", ns.connID, theirConnID)
```

Comment:
Are we switching away from logging this info in spans?

Reply:
For now, I think this fits better with what we're doing. 

I have a preliminary sketch from Jeff that outlines tracing and we could consider adding some of this data as attributes at that point.